x-pack/plugin/esql/compute/gen/src/main/java/org/elasticsearch/compute/gen/EvaluatorImplementer.java

```diff
@@ -182,8 +183,38 @@ private MethodSpec realEval(boolean blockStyle) {
         {
             processFunction.args.stream().forEach(a -> a.createScratch(builder));
 
+            boolean vectorize = false;
+            String vectorBuilderName = null;
+            if (blockStyle == false && processFunction.warnExceptions.isEmpty() && processOutputsMultivalued == false) {
+                ClassName type = processFunction.resultDataType(false);
+                vectorize = type.simpleName().startsWith("BytesRef") == false;
+                if (vectorize) {
+                    String arrayType = type.simpleName()
+                        .substring(0, type.simpleName().length() - "Vector".length())
+                        .toLowerCase(Locale.ROOT);
+                    builder.addStatement(arrayType + "[] buffer = result.values()");
+                }
+            }
             builder.beginControlFlow("position: for (int p = 0; p < positionCount; p++)");
-            {
+            if (vectorize) {
```

Comment:
Will this one ever be true?

Reply:
Line [190](https://github.com/elastic/elasticsearch/pull/109332/files/e03f318d2f1d8a379a86286d2d391dd8f7f6f47a#diff-f8d41f1685f1dc618e979b50575b98910207aa3794c18c0d5a7ec7c05202c684R190) set is to `true` for all the types apart from `BytesRef`. This is the core of the optimization